docs/INTERNALS.md

```
@@ -0,0 +1,686 @@
+| عدد (Int) | i64 |
+| عدد_عشري (Float) | double |
+| منطقي (Bool) | i1 |
+| نص (String) | %String* |
+| مصفوفة (Array) | %Array* |
+| Object | %ClassName* |
+
+### Class Layout
+
+Classes use a vtable-based dispatch:
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)